### PR TITLE
Add type cleanups

### PR DESCRIPTION
**Pull request by RobinNagpal**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: simulations/src/app/api/case-studies/[caseStudyId]/case-study-modules/[moduleId]/exercises/[exerciseId]/route.ts [hunk 1]
```diff
@@ -47,21 +18,20 @@ export interface UpdateModuleExerciseRequest {
...
   caseStudyId: string,
   moduleId: string,
   exerciseId: string,
   userId: string
-): Promise<ConsolidatedStudentExerciseResponse> {
-  // Get exercise with full context including all modules and exercises for navigation/progress calculations
-  const exerciseWithFullContext = await prisma.moduleExercise.findFirst({
+): Promise<ExerciseWithAttemptsResponse> {
+  // Get exercise with attempts
+  const exercise = await prisma.moduleExercise.findFirst({
```
Comment (RobinNagpal) [inline simulations/src/app/api/case-studies/[caseStudyId]/case-study-modules/[moduleId]/exercises/[exerciseId]/route.ts:29]: We can use findFirstOrThrow here to avoid extra if check

File: simulations/src/app/api/helpers/case-studies-util.ts [hunk 2]
```diff
@@ -378,9 +378,34 @@ export async function getStudentCaseStudy(caseStudyId: string, studentId: string
...
+
+      return {
+        ...exercise,
+        isExerciseCompleted,
+        attemptCount,
+      };
+    });
+
+    // Calculate module completion: all exercises must be completed
+    const isModuleCompleted = exercisesWithCalculatedData?.every((exercise) => exercise.isExerciseCompleted) || false;
```
Comment (RobinNagpal) [inline simulations/src/app/api/helpers/case-studies-util.ts:396]: Do we show tick on the module when the instructions have been read or when all exercises are completed?

File: simulations/src/lib/navigation-utils.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+/**
+ * Calculate navigation data for an exercise within a case study
+ * @param caseStudyData The full case study data with modules and exercises
+ * @param moduleId The current module ID
+ * @param exerciseId The current exercise ID
+ * @returns Navigation data including next/previous exercise information
+ */
+export function calculateNavigationData(
+  caseStudyData: CaseStudyWithRelationsForStudents | null | undefined,
```
Comment (RobinNagpal) [inline simulations/src/lib/navigation-utils.ts:11]: Probably the information passed should not be null or undefined? Do we need to fix the types?
